selfdrive/locationd/laikad.py

```diff
@@ -129,9 +130,27 @@ def read_report(self, gnss_msg):
       new_meas = read_raw_ublox(report)
     return week, tow, new_meas
 
+  def is_ephemeris(self, gnss_msg):
+    if self.use_qcom:
+      return gnss_msg.which == 'drSvPoly'
+    else:
+      return gnss_msg.which == 'ephemeris'
+
+  def read_ephemeris(self, gnss_msg):
+    # TODO this only works on GLONASS
+    if self.use_qcom:
+      # TODO this is not robust to gps week rollover
+      if self.gps_week is not None:
+        ephem = parse_qcom_ephem(gnss_msg.ephemeris, self.gps_week)
```

Comment:
this does not work, `gnss_msg` in qcom has the structure `drSvPoly` (see is_ephemeris) and this one does not contain an ephemeris field